R/Design-methods.R

```diff
@@ -1,165 +1,151 @@
-## Simulate outcomes from a CRM trial, assuming a true dose-toxicity
-## relationship.
-##
-## History:
-## 12/02/2014   file creation
-## 07/04/2014   start with parallelization on cores
-## 02/01/2015   rename: simulate.R --> Design-methods.R
-## 10/07/2015   added simulate methods
-#####################################################################################
-
-##' @include Data-methods.R
-##' @include Design-class.R
-##' @include McmcOptions-class.R
-##' @include Rules-methods.R
-##' @include Simulations-class.R
-##' @include helpers.R
-##' @include mcmc.R
-{}
-
-##' Helper function to set and save the RNG seed
-##'
-##' This is basically copied from simulate.lm
-##'
-##' @param seed an object specifying if and how the random number generator
-##' should be initialized (\dQuote{seeded}). Either \code{NULL} (default) or an
-##' integer that will be used in a call to \code{\link{set.seed}} before
-##' simulating the response vectors. If set, the value is saved as the
-##' \code{seed} slot of the returned object. The default, \code{NULL} will
-##' not change the random generator state.
-##' @return The RNGstate will be returned, in order to call this function
-##' with this input to reproduce the obtained simulation results
-##'
-##' @export
-##' @keywords programming
-##' @author Daniel Sabanes Bove \email{sabanesd@@roche.com}
-setSeed <- function(seed = NULL) {
+#' @include Data-methods.R
+#' @include Design-class.R
+#' @include McmcOptions-class.R
+#' @include Rules-methods.R
+#' @include Simulations-class.R
+#' @include helpers.R
+#' @include mcmc.R
+NULL
+
+# helper functions ----
+
+## set_seed ----
+
+#' Helper function to set and save the RNG seed
+#'
+#' @description `r lifecycle::badge("stable")`
+#'
+#' This code is basically copied from `stats:::simulate.lm`.
+#'
+#' @param seed an object specifying if and how the random number generator
+#' should be initialized (\dQuote{seeded}). Either `NULL` (default) or an
+#' integer that will be used in a call to [`set.seed`] before
+#' simulating the response vectors. If set, the value is saved as the
+#' `seed` slot of the returned object. The default, `NULL` will
+#' not change the random generator state.
+#' @return The RNGstate will be returned, in order to call this function
+#' with this input to reproduce the obtained simulation results.
+#'
+#' @export
+#' @keywords programming
+set_seed <- function(seed = NULL) {
   if (!exists(".Random.seed", envir = .GlobalEnv, inherits = FALSE)) {
     runif(1)
   }
 
   if (is.null(seed)) {
-    RNGstate <- get(".Random.seed", envir = .GlobalEnv)
+    rng_state <- get(".Random.seed", envir = .GlobalEnv)
   } else {
-    R.seed <- get(".Random.seed", envir = .GlobalEnv)
-    ## make sure R.seed exists in parent frame:
-    assign("R.seed", R.seed, envir = parent.frame())
+    r_seed <- get(".Random.seed", envir = .GlobalEnv)
+    # Make sure r_seed exists in parent frame.
+    assign("r_seed", r_seed, envir = parent.frame())
```

Comment:
improvement idea: let's call it
```suggestion
    assign(".r_seed", r_seed, envir = parent.frame())
```
so that it is hidden from the user

Reply:
Changed.